File: src/main/java/ru/ezhov/rocket/action/api/RocketActionSettings.java
package ru.ezhov.rocket.action.api;

import java.util.List;
import java.util.Map;

/**
 * Action settings
 */
public interface RocketActionSettings {
    /**
     * @return configurable action id
     */
    String id();

    /**
     * @return configurable action type
     */
    String type();

    /**
     * @return action settings
     */
    Map<String, String> settings();

    /**
     * @return configured child actions
     */
    List<RocketActionSettings> actions();
}

File: src/main/kotlin/ru/ezhov/rocket/action/api/RocketActionSettings.kt
package ru.ezhov.rocket.action.api

/**
 * Action settings
 */
interface RocketActionSettings {
    /**
     * @return configurable action id
     */
    fun id(): String

    /**
     * @return configurable action type
     */
    fun type(): String

    /**
     * @return action settings
     */
    fun settings(): Map<String, String>

    /**
     * @return configured child actions
     */
    fun actions(): List<RocketActionSettings>
}